lstchain/scripts/lstchain_mc_sensitivity.py

```diff
@@ -1,78 +1,86 @@
+    n_bins_energy = 20  #  Number of energy bins
+    n_bins_gammaness = 11  #  Number of gammaness bins
+    n_bins_theta2 = 10  #  Number of theta2 bins
     obstime = 50 * 3600 * u.s
     noff = 5
 
-    E, best_sens, result, units, gcut, tcut = sensitivity.find_best_cuts_sens(args.dl1file_gammas,
-                                                                              args.dl1file_protons,
-                                                                              args.dl2_file_g_cuts, args.dl2_file_p_cuts,
-                                                                              nfiles_gammas, nfiles_protons,
-                                                                              eb, gb, tb, noff,
-                                                                              obstime)
-    E, best_sens, result, units, dl2 = sensitivity.sens(args.dl1file_gammas,
-                                                        args.dl1file_protons,
-                                                        args.dl2_file_g_sens, args.dl2_file_p_sens,
-                                                        nfiles_gammas, nfiles_protons,
-                                                        eb, gcut, tcut * (u.deg ** 2), noff,
-                                                        obstime)
-    # plt.show()
-    plot_utils.sens_plot(eb, E, best_sens)
-    plt.show()
+    energy, best_sens, result, units, gcut, tcut = find_best_cuts_sensitivity(args.dl1file_gammas,
+                                               args.dl1file_protons,
+                                               args.dl2_file_g_cuts, args.dl2_file_p_cuts,
+                                               ntelescopes_gammas, ntelescopes_protons,
+                                               n_bins_energy, n_bins_gammaness, n_bins_theta2, noff,
+                                               obstime)
+
+    # For testing using fixed cuts
+    # gcut = np.ones(eb) * 0.8
+    # tcut = np.ones(eb) * 0.01
+
+    energy, best_sens, result, units, dl2 = sensitivity(args.dl1file_gammas,
+                                               args.dl1file_protons,
+                                               args.dl2_file_g_sens, args.dl2_file_p_sens,
+                                               nfiles_gammas, nfiles_protons,
+                                               eb, gcut, tcut * (u.deg**2), noff,
+                                               obstime)
 
     tab = Table.from_pandas(result)
 
     for i, key in enumerate(tab.columns.keys()):
         tab[key].unit = units[i]
-        if key == 'sensitivity':
+        if key=='sensitivity':
             continue
         tab[key].format = '8f'
 
     print(tab)
-    emed = np.sqrt(E[1:] * E[:-1])
 
-    plt.plot(emed[:-1], tab['hadron_rate'], label='Hadron rate', marker='o')
-    plt.plot(emed[:-1], tab['gamma_rate'], label='Gamma rate', marker='o')
+    plot_utils.sens_plot(eb, energy, best_sens)
```

Comment:
"sens_plot" is outdated.
sensitivity_plot_comparison?

Reply:
right